pkg/cmd/rollbackcopy/fileioutils.go

```diff
@@ -0,0 +1,152 @@
+package rollbackcopy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/klog"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+func dirExists(dirname string) bool {
+	info, err := os.Stat(dirname)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
+}
+
+func safeDirRename(src, dest string, srcMayNotExist bool) error {
+	if !dirExists(src) {
+		if !srcMayNotExist {
+			return fmt.Errorf("SafeDirRename: src dir %s does not exist", src)
+		} else {
+			return nil
+		}
+	}
+	destExists := false
+	destToBeRemoved := dest + ".to_be_removed"
+	if dirExists(dest) {
+		destExists = true
+		if dirExists(destToBeRemoved) {
+			os.RemoveAll(destToBeRemoved)
+		}
+		if err := os.Rename(dest, destToBeRemoved); err != nil {
+			return fmt.Errorf("safeDirRename: got error renaming %s %s. Err: %w", dest, destToBeRemoved, err)
+		}
+	}
+
+	if err := os.Rename(src, dest); err != nil {
+		if destExists {
+			os.Rename(destToBeRemoved, dest)
```

Comment:
Unhandled error

Reply:
This is any way error path, and the line below returns the real error that the caller needs. Returning error from this line is undesirable here.

Reply:
Logged the error.